src/documents/components/UploadSteps/NewExperienceStep.js

```diff
@@ -0,0 +1,292 @@
+import {Button, Grid, Image, Card, Icon, Checkbox} from 'semantic-ui-react';
+import documentText from '../../../assets/submit.png';
+import FlatFile from '../FlatFile/FlatFile';
+
+const NewExperienceStep = ({
+  nextStep,
+  previousStep,
+  createVersion,
+  updateVersion,
+  saveDocument,
+  createFlatfileSettings,
+  selectedTemplate,
+  version,
+  location,
+  history,
+  match,
+  values,
+  isSubmitting,
+  isValid,
+  setSubmitting,
+  setErrors,
+}) => {
+  const [flatfileSettings, setFlatfileSettings] = useState('');
+  const [source, setSource] = useState(null);
+  const [mappedData, setMappedData] = useState(null);
+  const [overwriteOld, setOverwriteOld] = useState(false);
+
+  const studyKfid = match.params.kfId;
+
+  if (location.state.file) {
+    const file = location.state.file;
+    var read = new FileReader();
+    read.readAsBinaryString(file);
+    read.onloadend = function() {
+      setSource(read.result);
+    };
+  }
+  console.log(overwriteOld);
+
+  useEffect(() => {
+    if (flatfileSettings.length === 0) {
+      createFlatfileSettings({
+        variables: {
+          templateVersions:
+            selectedTemplate.length > 0 ? [selectedTemplate] : [],
+        },
+      })
+        .then(resp => {
+          if (resp.data.createFlatfileSettings.flatfileSettings) {
+            setFlatfileSettings(
+              resp.data.createFlatfileSettings.flatfileSettings,
+            );
+          }
+        })
+        .catch(err => {
+          console.log(err);
+          const errorMessage = err.message
+            ? err.message
+            : 'Problem creating flatfile settings';
+          setFlatfileSettings('ERROR: ' + errorMessage);
+        });
+    }
+  }, [createFlatfileSettings, flatfileSettings.length, selectedTemplate]);
+
+  const handleSubmitNewDoc = () => {
+    const fileName = version.fileName || 'data.csv';
+    var file = new File([mappedData], fileName, {
+      type: 'text/plain',
+    });
+    createVersion({
+      variables: {
+        file: file,
+        description: 'Mapped version',
+        study: Buffer.from('StudyNode:' + match.params.kfId).toString('base64'),
+      },
+    })
+      .then(resp => {
+        saveDocument({
+          variables: {
+            version: resp.data.createVersion.version.id,
+            study: Buffer.from('StudyNode:' + match.params.kfId).toString(
+              'base64',
+            ),
+            name: values.file_name,
+            fileType: selectedTemplate ? 'OTH' : values.file_type,
+            description: values.file_desc,
+            tags: [],
+            templateVersion:
+              selectedTemplate.length > 0 ? selectedTemplate : null,
+          },
+        })
+          .then(resp => {
+            if (!overwriteOld) {
+              updateVersion({
+                variables: {
+                  versionId: version.kfId,
+                  document: resp.data.createFile.file.id,
+                  description: 'Original upload',
```

Comment:
It looks like these descriptions are reversed. The file that gets described as Original upload, seems to be the one that came out of mapping. And vice versa. The original file that didn't get mapped, is labeled as Mapped version